{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential Expression with DeSeq2\n",
    "\n",
    "Now that I have my [count data](https://github.com/yaaminiv/yaaminiv-fish546-2016/blob/master/notebooks/2016-11-04-oly-gonad-OA-part3-kallisto.ipynb), I can use [DeSeq2](https://github.com/yaaminiv/yaaminiv-fish546-2016/blob/master/tutorials/DESeq2-tutorial/2016-10-26-DESeq2-Tutorial-Part-2.ipynb) to analyze differential expression. The process I used was adapted from [this notebook](https://github.com/sr320/eimd-sswd/blob/master/eimd_analysis.ipynb).\n",
    "\n",
    "My goal is to compare gene expression between the female-106 and female-108 samples, and between the male-106 and male-108 samples. To do this, I will use the `DeSeq2` package in R. Before this can happen, I need to make sure my data is in the right format for analysis.\n",
    "\n",
    "### 1. Reformat `kallisto quant` output files\n",
    "\n",
    "#### Convert to a `.txt` file\n",
    "\n",
    "My first step is converting my count data into a .txt file to use in DeSeq2. This is as simple as opening the `.tsv` file generated by `kallisto quant` in excel, and saving it as a `.txt` file."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "![example file conversion](https://raw.githubusercontent.com/yaaminiv/yaaminiv-fish546-2016/master/analyses/kallisto-female-108/samplefileconversion.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Merge count data files\n",
    "\n",
    "Next, I need to merge all of my count data files so I have one `.txt` file with all of my count information (the column label in the original files have the word \"count\" in them). The column headers will be as follows:\n",
    "\n",
    "- Feature_ID\n",
    "- Female_106 - Total gene reads\t\n",
    "- Male_106 - Total gene reads\t\n",
    "- Female_108 - Total gene reads\t\n",
    "- Male_108 - Total gene reads\n",
    "\n",
    "I merged these in Excel. Additionally, I converted all of my count data to integers, as DESeq2 will only work with nonnegative integers. My final file looks like this:\n",
    "\n",
    "![oly-gonad-oa-counts]()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Use DeSeq2 in R\n",
    "\n",
    "I am now ready to complete my analyses in R! R scripts for the four different pairwise comparisons I tested can be viewed below:\n",
    "\n",
    "#### Control vs. Ocean Acidification conditions\n",
    "\n",
    "[female-106 vs. female-108]()\n",
    "![female-106 vs. female-108]()\n",
    "\n",
    "[male-106 vs. male-108]()\n",
    "![male-106 vs. male-108]()\n",
    "\n",
    "#### Male vs. Female Gonads\n",
    "\n",
    "[female-106 vs. male-106]()\n",
    "![female-106 vs. male-106]()\n",
    "\n",
    "[female-108 vs. male-108]()\n",
    "![female-108 vs. male-108]()"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
